### PR TITLE
chore(medusa): Clean up atomicPhase usage

### PR DESCRIPTION
**Pull request by srindom**
**What**
Cleans up `atomicPhase` usage as part of our approach to solving #1834

### CODE REVIEW

File: packages/medusa/src/services/discount.ts [hunk 1]
```diff
@@ -562,9 +550,11 @@ class DiscountService extends TransactionBaseService<DiscountService> {
         return false
       }
 
-      const product = await this.productService_.retrieve(productId, {
-        relations: ["tags"],
-      })
+      const product = await this.productService_
```
Comment (adrien2p) [inline packages/medusa/src/services/discount.ts:553]: Praise: well spotted :)

File: packages/medusa/src/services/discount.ts [hunk 2]
```diff
@@ -721,24 +711,23 @@ class DiscountService extends TransactionBaseService<DiscountService> {
...
+    // Instead of throwing on missing customer id, we simply invalidate the discount
+    if (!customerId) {
+      return false
+    }
 
-      return await discountConditionRepo.canApplyForCustomer(
-        discountRuleId,
-        customer.id
-      )
+    const customer = await this.customerService_.retrieve(customerId, {
```
Comment (adrien2p) [inline packages/medusa/src/services/discount.ts:723]: todo: we should pass manager to a with transaction here

File: packages/medusa/src/services/gift-card.ts [hunk 3]
```diff
@@ -253,7 +245,9 @@ class GiftCardService extends TransactionBaseService<GiftCardService> {
       const { region_id, metadata, balance, ...rest } = update
 
       if (region_id && region_id !== giftCard.region_id) {
-        const region = await this.regionService_.retrieve(region_id)
+        const region = await this.regionService_
```
Comment (adrien2p) [inline packages/medusa/src/services/gift-card.ts:248]: 👍

File: packages/medusa/src/services/price-list.ts [hunk 4]
```diff
@@ -328,43 +326,42 @@ class PriceListService extends TransactionBaseService<PriceListService> {
...
-
-                return productVariantRepo.create({
-                  ...v,
-                  prices,
-                })
+    const manager = this.manager_
+    const productVariantRepo = manager.getCustomRepository(
+      this.productVariantRepo_
+    )
+    const [products, count] = await this.productService_.listAndCount(
```
Comment (adrien2p) [inline packages/medusa/src/services/price-list.ts:333]: todo: we should pass manager to a with transaction here

File: packages/medusa/src/services/price-list.ts [hunk 5]
```diff
@@ -377,29 +374,28 @@ class PriceListService extends TransactionBaseService<PriceListService> {
...
-
-      const variantsWithPrices = await Promise.all(
-        variants.map(async (variant) => {
-          const [prices] = await moneyAmountRepo.findManyForVariantInPriceList(
-            variant.id,
-            priceListId,
-            requiresPriceList
-          )
+    const manager = this.manager_
+    const [variants, count] = await this.variantService_.listAndCount(
```
Comment (adrien2p) [inline packages/medusa/src/services/price-list.ts:378]: todo: we should pass manager to a with transaction here

File: packages/medusa/src/services/pricing.ts [hunk 6]
```diff
@@ -63,34 +63,28 @@ class PricingService extends TransactionBaseService<PricingService> {
...
-            ...context,
-            currency_code: currencyCode,
-          },
-          automatic_taxes: automaticTaxes,
-          tax_rate: taxRate,
-        }
-      }
-    )
+    if (context.region_id) {
+      const region = await this.regionService.retrieve(context.region_id, {
```
Comment (adrien2p) [inline packages/medusa/src/services/pricing.ts:71]: todo: we should pass manager to a with transaction here

**Review by adrien2p — COMMENTED** [review]
Few missing transactions to be pass down but otherwise it is LGTM for me :)

**Review by olivermrbl — APPROVED** [review]
LGTM, super nice!

File: packages/medusa/src/services/customer.ts [hunk 7]
```diff
@@ -139,30 +137,28 @@ class CustomerService extends TransactionBaseService<CustomerService> {
...
    * @return {Promise} the result of the count operation
    */
   async count(): Promise<number> {
-    return await this.atomicPhase_(async (manager) => {
-      const customerRepo = manager.getCustomRepository(this.customerRepository_)
-      return await customerRepo.count({})
-    })
+    const manager = this.manager_
+    const customerRepo = manager.getCustomRepository(this.customerRepository_)
+    return await customerRepo.count({})
```
Comment (adrien2p) [inline packages/medusa/src/services/customer.ts:161]: question(non blocking): why are we passing an empty object here? the options are optional and I would have assume that no arguments would give the same result no?

**Review by adrien2p — APPROVED** [review]
LGTM 🎉

**Pull request merged**